workflow-service/src/test/java/com/redhat/parodos/workflow/execution/repository/WorkFlowRepositoryTest.java

```diff
@@ -124,6 +124,37 @@ public void testFindByStatusInAndIsMain() {
 		assertEquals(mainWorkFlowExecution, workFlowExecutions.get(0));
 	}
 
+	@Test
+	public void testCountRestartedWorkflow() {
+		// given
+		User user = createUser();
+		WorkFlowExecution mainWorkFlowExecution = createWorkFlowExecution();
+		WorkFlowExecution workFlowExecution = WorkFlowExecution.builder().workFlowDefinition(createWorkFlowDefinition())
+				.status(WorkStatus.IN_PROGRESS).projectId(createProject(user).getId()).user(createUser())
+				.mainWorkFlowExecution(mainWorkFlowExecution).build();
+		workFlowRepository.save(workFlowExecution);
+
+		// when
+		int count = workFlowRepository.countRestartedWorkflow(workFlowExecution.getId());
+
+		// then
+		assertEquals(0, count);
+
+		// When
+		WorkFlowExecution restartedWorkFlowExecution = WorkFlowExecution.builder()
+				.workFlowDefinition(createWorkFlowDefinition()).status(WorkStatus.IN_PROGRESS)
+				.projectId(createProject(user).getId()).user(createUser()).mainWorkFlowExecution(mainWorkFlowExecution)
+				.originalWorkFlowExecution(workFlowExecution).build();
+		workFlowRepository.save(restartedWorkFlowExecution);
+
+		count = workFlowRepository.countRestartedWorkflow(workFlowExecution.getId());
+		int countForRestarted = workFlowRepository.countRestartedWorkflow(restartedWorkFlowExecution.getId());
+
+		// then
+		assertEquals(1, count);
+		assertEquals(0, countForRestarted);
```

Comment:
This is interesting: should we allow a restart of a restarted workflow or refer the caller to its original?

Reply:
IMO we should keep the workflow simple: the original workflow execution ID is the one provided by the user, we do not go up to get the original (original)* workflow execution ID. And if we allow partial argument overriding the allows us to keep trace of the possible changes step by step if needed